File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/filesystem/FileSystemScriptConfig.java
package com.sequenceiq.cloudbreak.service.cluster.flow.filesystem;

import java.util.HashMap;
import java.util.Map;

import com.sequenceiq.cloudbreak.api.model.ExecutionType;
import com.sequenceiq.cloudbreak.service.cluster.flow.ClusterLifecycleEvent;

public class FileSystemScriptConfig {
    private String scriptLocation;

    private ClusterLifecycleEvent clusterLifecycleEvent;

    private ExecutionType executionType;

    private Map<String, String> properties = new HashMap<>();

    public FileSystemScriptConfig(String scriptLocation, ClusterLifecycleEvent clusterLifecycleEvent, ExecutionType executionType) {
        this.scriptLocation = scriptLocation;
        this.clusterLifecycleEvent = clusterLifecycleEvent;
        this.executionType = executionType;
    }

    public FileSystemScriptConfig(String scriptLocation, ClusterLifecycleEvent clusterLifecycleEvent,
            ExecutionType executionType, Map<String, String> properties) {
        this(scriptLocation, clusterLifecycleEvent, executionType);
        this.properties = properties;
    }

    public String getScriptLocation() {
        return scriptLocation;
    }

    public ClusterLifecycleEvent getClusterLifecycleEvent() {
        return clusterLifecycleEvent;
    }

    public ExecutionType getExecutionType() {
        return executionType;
    }

    public Map<String, String> getProperties() {
        return properties;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/filesystem/FileSystemScriptConfig.kt
package com.sequenceiq.cloudbreak.service.cluster.flow.filesystem

import java.util.HashMap

import com.sequenceiq.cloudbreak.api.model.ExecutionType
import com.sequenceiq.cloudbreak.service.cluster.flow.ClusterLifecycleEvent

class FileSystemScriptConfig(val scriptLocation: String, val clusterLifecycleEvent: ClusterLifecycleEvent, val executionType: ExecutionType) {
    val properties: Map<String, String> = HashMap()

    constructor(scriptLocation: String, clusterLifecycleEvent: ClusterLifecycleEvent,
                executionType: ExecutionType, properties: Map<String, String>) : this(scriptLocation, clusterLifecycleEvent, executionType) {
        this.properties = properties
    }
}
